src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -402,11 +402,42 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 	List *originalForeignKeyRecreationCommands = NIL;
 	if (IsCitusTableType(relationId, CITUS_LOCAL_TABLE))
 	{
+		char *distributionColumnName = NULL;
+
+		if (distributionMethod != DISTRIBUTE_BY_NONE)
+		{
+			/* save the column name to update the distribution column var later */
+			distributionColumnName =
+				ColumnToColumnName(relationId, nodeToString(distributionColumn));
```

Comment:
is there a way to avoid passing a Var to this function, but instead pass the name and resolve the name after these steps?

Reply:
What Marco (talked offline) actually means is that we should pass distribution column name (string) to `CreateDistributedTable`, and then create the Var distribution column in `CreateDistributedTable`.

I've paused this pr and currently working on #5723 for this approach